hours/models.py

```diff
@@ -667,6 +667,9 @@ def as_hash_input(self) -> str:
     def as_text(self) -> str:
         group_strings = []
         for time_span_group in self.time_span_groups.all():
+            # we don't know why there are empty time_spans vector, this is a workaround for the time being
+            if len (time_span_group.time_spans.all()) == 0:
+                continue
             if time_span_group.is_removed:
                 continue
```

Comment:
I would first check if the group is removed, only then check if the group has any time spans.

Also in Django the correct way to check if there are items in a queryset is with count. i.e.:

```python
if not time_span_group.time_spans.count():
```

Commit message could be changed too... We know that there can be empty time span groups. If someone removes all time spans from a group the group is not automatically removed.

Reply:
Oh nice! Thanks. I'll make the changes.